java/serving/src/main/java/feast/serving/service/OnlineServingServiceV2.java

```diff
@@ -110,13 +112,33 @@ public GetFeastServingInfoResponse getFeastServingInfo(
       throw new RuntimeException("Requested features list must not be empty");
     }
 
+    List<EntityProto.Entity> entities = this.registryRepository.getEntities();
+    Map<String, List<String>> entityNamesPerFeatureView =
+        retrievedFeatureReferences.stream()
+            .map(this.registryRepository::getFeatureViewSpec)
+            .distinct()
+            .collect(
+                Collectors.toMap(
+                    FeatureViewSpec::getName,
+                    spec ->
+                        spec.getEntitiesList().stream()
+                            .map(
+                                name ->
+                                    entities.stream()
```

Comment:
Since it's latency sensitive part I would suggest to replace this lookup with Map<EntityName -> JoinKey> that will be provided by registry, for example